### PR TITLE
[Tabular][fastai] Missing category fix

### PR DESCRIPTION
**Pull request by gradientsky**
*Description of changes:*
- fixing missing category handling in fastai model

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py [hunk 1]
```diff
@@ -146,8 +147,10 @@ def preprocess_train(self, X_train, y_train, X_val, y_val, **kwargs):
         return data
 
     def fill_missing(self, df):
-        df = df[self.cat_columns + self.cont_columns]
+        df = df[self.cat_columns + self.cont_columns].copy()
         for c in self.cat_columns + self.cont_columns:
+            if c in self.cat_columns:
+                df[c] = df[c].cat.add_categories(MISSING)
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py:153]: What happens if `MISSING` is already is a category?

Comment (gradientsky) [reply tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py:153]: We need to pick something: this category is supposed to be something not in the training set by design.

**Comment by szha** [comment]
Job PR-844-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-844/1/index.html

**Comment by szha** [comment]
Job PR-844-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-844/3/index.html

File: tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py (hunk 1 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py:153]: The comment wasn't about changing the missing value, it was to ensure that we didn't crash if it was present.

**Comment by szha** [comment]
Job PR-844-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-844/4/index.html

**Comment by Innixma** [comment]
No issues found in benchmarking, looks good!

**Review by Innixma — APPROVED** [review]

**Pull request merged**